src/Tar.jl

```diff
@@ -461,4 +481,24 @@ check_tree_hash_tarball(tarball::AbstractString) =
 
 check_tree_hash_tarball(tarball::ArgRead) = nothing
 
+function check_windows_path(
+    path  :: AbstractString,
+    parts :: AbstractVector{<:AbstractString} = split(path, r"/+"),
+)
+    for part in parts
+        isempty(part) && continue
+        ch = part[end]
+        if ch == ' ' || firstindex(part) < lastindex(part) && ch == '.'
+            error("illegal Windows char: $(repr(ch)) in $(repr(path))")
+        end
+        for ch in part
+            ch ≥ ' ' && ch ∉ "\"*:<>?\\|" && continue
+            error("illegal Windows char: $(repr(ch)) in $(repr(path))")
+        end
+        if occursin(r"^(CON|PRN|AUX|NUL|(COM|LPT)[1-9])(\.|$)"i, part)
```

Comment:
I think we need to turn off UTF and UCP flags, to use CASELESS for this purpose?

Or write this as `"^([Cc][Oo[Nn]|[Pp][Rr][Nn]|` and so on?

Reply:
I'm not quite following you here. Are you talking about PRCE flags or something else. Is the worry about invalid Unicode path names? In that case maybe it would be better to require a valid name in the first place? After all, if the path is malformed UTF-8 how would we even create that name on Windows, which uses UTF-16? We *could* still allow malformed but invalid UTF-8, but why bother?

Reply:
I thought CASELESS is locale-sensitive, unless you force it into ASCII mode, but I am not certain

Reply:
I rather doubt there are any locales where ASCII case behaves differently, but I suppose it's possible. I also didn't think that we respect locales in this respect (we shouldn't—they're a bad idea).

Reply:
> When caseless matching is specified (the PCRE2_CASELESS option or (?i) within the pattern), letters are matched independently of case. Note that there are two ASCII characters, K and S, that, in addition to their lower case ASCII equivalents, are case-equivalent with Unicode U+212A (Kelvin sign) and U+017F (long S) respectively when either PCRE2_UTF or PCRE2_UCP is set.

https://www.pcre.org/current/doc/html/pcre2pattern.html#SEC9

I guess we might escape this being a problem because neither K nor S are part of the set we need to match? And as long as we don't call `pcre2_set_character_tables` it won't enable any other locale's tables?

https://www.pcre.org/current/doc/html/pcre2api.html#SEC22